### PR TITLE
fix(auth): force update default mqtt_user when password or hash_type changed.

### PR DESCRIPTION
**Pull request by zhongwencool**
ISSUE:
Some users have changed the user password or hash_type in the configuration file(`plugins/emqx_auth_mnesia.conf`), but the reboot never takes effect.
```bash
auth.mnesia.1.login = admin
auth.mnesia.1.password = public
## Value: plain | md5 | sha | sha256
auth.mnesia.password_hash = sha256
```
we need update default value in `emqx_user` table when the password not match, then print a warning log "password has be updated".

By the way:
if the user changes the password_hash type, we can't find a good way to change the existing non-default user, so we have to let the user update it manually.

### CODE REVIEW

File: apps/emqx_auth_mnesia/src/emqx_auth_mnesia.appup.src [hunk 1]
```diff
@@ -7,10 +7,14 @@
       {update, emqx_auth_mnesia_sup, supervisor},
       {apply, {emqx_acl_mnesia_migrator, start_supervised, []}},
       {load_module,emqx_auth_mnesia_api, brutal_purge,soft_purge,[]},
-      {load_module,emqx_acl_mnesia, brutal_purge,soft_purge,[]},
+      {load_module,emqx_acl_mnesia, brutal_purge, soft_purge,[]},
```
Comment (HJianBo) [inline apps/emqx_auth_mnesia/src/emqx_auth_mnesia.appup.src:10]: A bit of a strange space character

File: apps/emqx_auth_mnesia/src/emqx_auth_mnesia_cli.erl [hunk 2]
```diff
@@ -56,6 +57,28 @@ insert_user(User = #emqx_user{login = Login}) ->
...
+
+insert_or_update_user(NewPwd, User = #emqx_user{login = Login}) ->
+    case mnesia:read(?TABLE, Login) of
+        []    -> mnesia:write(User);
+        [#emqx_user{password = Pwd}] ->
+            case emqx_auth_mnesia:match_password(NewPwd, hash_type(), [Pwd])  of
+                true -> ok;
+                false ->
+                    Res = mnesia:write(User),
+                    ?LOG(warning, "[Mnesia] (~p)'s password has be updated.", [Login]),
```
Comment (zmstone) [inline apps/emqx_auth_mnesia/src/emqx_auth_mnesia_cli.erl:76]: move the log out of mnesia transaction.

**Comment by zmstone** [comment]
update CHANGES-4.3.md

**Comment by zhongwencool** [comment]
> update CHANGES-4.3.md

https://github.com/emqx/emqx-docs/pull/983

**Review by HJianBo — DISMISSED** [review]

**Review by zmstone — APPROVED** [review]

**Pull request merged**